app/src/test/java/de/westnordost/streetcomplete/quests/cycleway/AddCyclewayTest.kt

```diff
@@ -208,4 +208,18 @@ class AddCyclewayTest {
         assertEquals(0, questType.getApplicableElements(mapData).toList().size)
         assertFalse(questType.isApplicableTo(residentialWayIn30Zone)!!)
     }
+
+    @Test
+    fun `not applicable to residential road with maxspeed 30`() {
+        val residentialWayWithMaxspeed30 = way(1L, listOf(1, 2, 3), mapOf(
+            "highway" to "residential",
+            "maxspeed" to "30",
+        ))
+
+        val mapData = TestMapDataWithGeometry(listOf(residentialWayWithMaxspeed30))
+
+        // a residential way with maxspeed=30 and no other maxspeed tags is assumed to be in a max-speed 30 zone
```

Comment:
What about the following way:
```
highway=residential
maxspeed=30
zone:traffic=DE:urban
```

Right now it will be applicable. Is this intended?

Reply:
Yes, because (in Germany) only in 30 **zones** the traffic agency is not allowed to establish (benutzungspflichtige) bike paths. (Or actually, the other way round, it can only be a 30 zone if there are no benutzungspflichtige bike paths)